marketplace/deployer_util/config_helper.py

```diff
@@ -778,6 +778,28 @@ class SchemaXServiceAccount:
 
   def __init__(self, dictionary):
     self._roles = dictionary.get('roles', [])
+    for role in self._roles:
+      if role.get('rulesType') == 'PREDEFINED':
+        if role.get('rules'):
+          raise InvalidSchema('rules can only be used with rulesType CUSTOM')
+        if not role.get('rulesFromRoleName'):
+          raise InvalidSchema('Missing rulesFromRoleName for PREDEFINED role')
+      elif role.get('rulesType') == 'CUSTOM':
+        if role.get('rulesFromRoleName'):
+          raise InvalidSchema(
+              'rulesFromRoleName can only be used with rulesType PREDEFINED')
+        if not role.get('rules'):
+          raise InvalidSchema('Missing rules for CUSTOM role')
+        for rule in role.get('rules', []):
+          if rule.get('nonResourceURLs'):
```

Comment:
probably require `resources` as well, since otherwise the rules don't really have effects

Reply:
Good call, added check for `resources` and `verbs`. I had forgotten to do so since no one's made that mistake yet - now no one will :)